drivers/i2c/i2c_ll_stm32.meta.jinja2

```diff
@@ -0,0 +1,36 @@
+{# Copyright (c) 2018 Nordic Semiconductor ASA #}
+{# SPDX-License-Identifier: Apache-2.0         #}
+
+{% extends 'device_declare.jinja2' %}
+
+{% block init_params %}
+        {% do params.update({'irq_flag':'CONFIG_I2C_STM32_INTERRUPT'}) %}
+        {% do params.update({'config_struct':'i2c_stm32_config'}) %}
+        {% do params.update({'data_struct':'i2c_stm32_data'}) %}
+        {% do params.update({'api_struct':'api_funcs'}) %}
+        {% do params.update({'init_function':'i2c_stm32_init'}) %}
+{% endblock %}
+
+{# Because these blocks are "scoped" all variables in the macro from the
+   base template can be accessed in these blocks #}
+{% block config_struct_body %}
+        .i2c = (I2C_TypeDef *){{ "%x"|format(device['reg']['0']['address']) }},
```

Comment:
Would you be able to use recently introduced  get_reg functions ?
@galak 
What about doing the formatting inside get_reg directly  ?

Reply:
I agreed, can we use the edtsdevice get_reg() api.

@erwango hmm, I want to keep the api's as actual integers.  Wonder if {{ hex(device['reg']['0']['address']) }} works

Reply:
yes, I can make any python API available as a data object in the template or custom filters. 

@galak If you see in template code, I use the format filter to turn an integer into a hex value no need to use an external function

Reply:
@evenl , aim of my comment was to find a way provide a more simple syntax instead of
```
{{ "%x"|format(device['reg']['0']['address']) }}
```

@galak: what about get_hex_reg()?

Reply:
@erwango I was hoping to not add such APIs, and thus wondered if the following would work: `{{ hex(device['reg']['0']['address']) }}`